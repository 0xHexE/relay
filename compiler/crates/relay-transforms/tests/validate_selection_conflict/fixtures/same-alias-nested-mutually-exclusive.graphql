# expected-to-throw
fragment TestFragment on User {
  nameRenderer {
    ...frag1
    ...frag2

  }
}

fragment frag1 on PlainUserNameRenderer {
  data {
    text
  }
}

fragment frag2 on CustomNameRenderer {
  data: user {
    text: canViewerLike
  }
}
