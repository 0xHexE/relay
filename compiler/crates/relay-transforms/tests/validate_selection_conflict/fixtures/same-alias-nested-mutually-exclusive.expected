==================================== INPUT ====================================
# expected-to-throw
fragment TestFragment on User {
  nameRenderer {
    ...frag1
    ...frag2

  }
}

fragment frag1 on PlainUserNameRenderer {
  data {
    text
  }
}

fragment frag2 on CustomNameRenderer {
  data: user {
    text: canViewerLike
  }
}
==================================== ERROR ====================================
✖︎ Field 'text' is ambiguous because it references fields with different types: 'text' with type 'String' and 'canViewerLike' with type 'Boolean'

  same-alias-nested-mutually-exclusive.graphql:12:5
   11 │   data {
   12 │     text
      │     ^^^^
   13 │   }

  ℹ︎ the other field

  same-alias-nested-mutually-exclusive.graphql:18:11
   17 │   data: user {
   18 │     text: canViewerLike
      │           ^^^^^^^^^^^^^
   19 │   }
